src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -134,6 +153,27 @@ async def ensure_executor_health(self):
             retry_count += 1
         raise ExecutorServiceUnhealthy(f"{EXECUTOR_UNHEALTHY_MESSAGE}. Please resubmit your flow and try again.")
 
+    async def _check_health(self):
+        try:
+            health_url = self.api_endpoint + "/health"
+            async with httpx.AsyncClient() as client:
+                response = await client.get(health_url)
+            if response.status_code != 200:
+                bulk_logger.warning(f"{EXECUTOR_UNHEALTHY_MESSAGE}. Response: {response.status_code} - {response.text}")
+                return False
+            return True
+        except Exception as e:
+            bulk_logger.warning(f"{EXECUTOR_UNHEALTHY_MESSAGE}. Error: {str(e)}")
+            return False
+
+    def _check_startup_error_from_file(self, error_file) -> Exception:
+        error_dict = load_json(error_file)
+        if error_dict:
+            bulk_logger.error(f"Error when starting the executor service: {error_dict}")
```

Comment:
Shall we format the error dict using `json.dumps(indent=4)` to improve readability? (Does the App Insights support multiline logs?)

Reply:
Updated. Only print message and innermost_error_code.